homeassistant/components/opentherm_gw/__init__.py

```diff
@@ -364,6 +375,18 @@ async def set_led_mode(call: ServiceCall) -> None:
         DOMAIN, SERVICE_SET_LED_MODE, set_led_mode, service_set_led_mode_schema
     )
 
+    async def set_max_ch_setpoint(call: ServiceCall) -> None:
+        """Set the maximum central heating setpoint on the OpenTherm Gateway."""
+        gw_dev = hass.data[DATA_OPENTHERM_GW][DATA_GATEWAYS][call.data[ATTR_GW_ID]]
+        await gw_dev.gateway.set_max_ch_setpoint(call.data[ATTR_TEMPERATURE])
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_SET_MAX_CH_SETPOINT,
+        set_max_ch_setpoint,
+        service_set_max_ch_setpoint_schema,
+    )
```

Comment:
Why isn't  this just an entity?

Reply:
I relied on the set_control_setpoint service above.  @mvn23 will be able to answer better.  But I'm afraid I don't understand: can an entity set a value on a device?

Reply:
Upon reflection, I think a service is better than an entity.  Indeed, the services are pretty much hidden for ordinary users.  Changing boiler temperatures is intended for a well-informed people as mentioned in the hardware [docs](http://otgw.tclcode.com/standalone.html#considerations). An entity would make a slider or box easily accessible in the interface, which is undesirable, although of course it can be hidden by default.
But perhaps @mvn23  as code owner has an another opinion.

Reply:
Unfortunately my current knowledge does not allow me to create a configuration entity that interacts with the pyotgw library. Too bad, but for my part, I will continue to use this service in custom_components, until better.

Reply:
That's too bad, but thank you for your effort anyway.

@home-assistant close

Reply:
> That's too bad, but thank you for your effort anyway.

You're welcome.

P.S. I'm preparing another PR to remove obsolete lines as suggested above.